pkg/process/checks/chunking.go

```diff
@@ -12,136 +12,41 @@ import (
 	// containers seen within the same `model.ContainerProc` as processes.
-	chunker.container = ctr
-	list := &processList{
-		procs:   procs,
-		chunker: chunker,
-	}
-	util.ChunkPayloadsBySizeAndWeight(list, chunker, maxChunkSize, maxChunkWeight)
-}
-
-// processList is a payload list of `model.Process` payloads
-type processList struct {
-	procs   []*model.Process
-	chunker processChunker
-}
-
-// processChunker abstracts chunking of `model.Process` payloads
-type processChunker interface {
-	// Accept takes a slice of `model.Process` and allocates them to the current chunk
-	Accept(procs []*model.Process, weight int)
-}
-
-func (l *processList) Len() int {
-	return len(l.procs)
-}
-
-func (l *processList) WeightAt(idx int) int {
-	if idx >= len(l.procs) {
-		return 0
-	}
-	return weighProcess(l.procs[idx])
-}
-
-func (l *processList) ToChunk(start, end int, weight int) {
-	l.chunker.Accept(l.procs[start:end], weight)
-}
-
-// chunkProps is used to track weight and size of chunks
-type chunkProps struct {
-	weight int
-	size   int
-}
-
-// chunkPropsTracker tracks weight and size of chunked payloads
-type chunkPropsTracker struct {
-	props []chunkProps
-	idx   int
-}
-
-// TakenSize returns the size allocated to the current chunk
-func (c *chunkPropsTracker) TakenSize() int {
-	if c.idx < len(c.props) {
-		return c.props[c.idx].size
-	}
-	return 0
-}
-
-// TakenWeight returns the weight allocated to the current chunk
-func (c *chunkPropsTracker) TakenWeight() int {
-	if c.idx < len(c.props) {
-		return c.props[c.idx].weight
-	}
-	return 0
-}
-
-// Append creates a new chunk at the end (cases when it is known any previously allocated chunks cannot fit the payload)
-func (c *chunkPropsTracker) Append() {
-	c.idx = len(c.props)
-}
-
-// Next moves to the next chunk or allocates a new chunk if the current chunk is the last
-func (c *chunkPropsTracker) Next() {
-	c.idx++
-}
-
-// collectorProcChunker implements allocation of chunks to `model.CollectorProc`
-type collectorProcChunker struct {
-	chunkPropsTracker
-	container      *model.Container
-	collectorProcs []*model.CollectorProc
-}
-
-// collectprProcChunker implements both `chunkAllocator` and `processChunker`
-var _ processChunker = &collectorProcChunker{}
-var _ util.ChunkAllocator = &collectorProcChunker{}
-
-func (c *collectorProcChunker) Accept(procs []*model.Process, weight int) {
-	if c.idx >= len(c.collectorProcs) {
-		// If we are outside of the range of allocated chunks, allocate a new one
-		c.collectorProcs = append(c.collectorProcs, &model.CollectorProc{})
-		c.props = append(c.props, chunkProps{})
-	}
-
-	collectorProc := c.collectorProcs[c.idx]
-
-	// Note that we are currently not accounting for the container size/weight in calculations
-	if c.container != nil {
-		collectorProc.Containers = append(collectorProc.Containers, c.container)
+	chunker.OnAccept = func(t *model.CollectorProc) {
```

Comment:
Why not make this a method? It doesn't look like this field is ever assigned elsewhere.

Reply:
I'm not quite following what you mean here. OnAccept is used inside of the core chunking lib.

Reply:
Spoke offline, atm there's not really a great way to refactor this without tearing out a lot of existing code. I think it should be fine to keep it as it is, as long as there's a good explanation for what is going here in the comments.